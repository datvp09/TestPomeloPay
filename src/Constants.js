import {Platform, Dimensions} from 'react-native';
import {isIphoneX} from 'react-native-iphone-x-helper';

const {height, width} = Dimensions.get('window');
const isiOS = Platform.OS === 'ios';
const isIpX = isIphoneX(); // X, XS, XR
const isSmallScreen = height < 650;
const isBigScreen = isiOS ? height > 800 : height > 730;
const intenseBlue = '#1365AF';
const refundIcon = require('./assets/images/iconBack.png');

export {
  height,
  width,
  isiOS,
  isIpX,
  isSmallScreen,
  isBigScreen,
  intenseBlue,
  refundIcon,
};
