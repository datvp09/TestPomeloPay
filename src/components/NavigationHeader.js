import React, {Component} from 'react';
import {View, Text, StyleSheet, StatusBar} from 'react-native';
import {isIpX, isiOS, isSmallScreen, intenseBlue} from '../Constants';
import PropTypes from 'prop-types';
import {getStatusBarHeight} from 'react-native-iphone-x-helper';

const APPBAR_HEIGHT = isSmallScreen ? 30 : 35;
const STATUSBAR_HEIGHT = isiOS
  ? isIpX
    ? getStatusBarHeight([true])
    : 20
  : StatusBar.currentHeight;

export const headerHeight = isIpX
  ? STATUSBAR_HEIGHT + 5
  : APPBAR_HEIGHT + STATUSBAR_HEIGHT;

export default class NavigationHeader extends Component {
  renderTitle = () => {
    const {title, titleStyle} = this.props;

    if (!title) {
      return null;
    }

    if (typeof title === 'string') {
      return <Text style={[styles.titleText, titleStyle]}>{title}</Text>;
    }

    return title();
  };

  renderRightView = () => {
    const {rightView} = this.props;

    if (rightView) {
      return rightView();
    }

    return <View style={styles.rightView} />;
  };

  renderStatusBarIOS = () => {
    if (!isiOS) {
      return null;
    }

    if (isIpX) {
      return (
        <View
          style={{
            height: getStatusBarHeight([true]),
            backgroundColor: intenseBlue,
          }}
        />
      );
    }
    return <View style={{height: 20, backgroundColor: intenseBlue}} />;
  };

  render() {
    const {style} = this.props;

    return (
      <View>
        {this.renderStatusBarIOS()}
        <View style={[styles.container, style]}>
          {this.renderTitle()}
          {this.renderRightView()}
        </View>
      </View>
    );
  }
}

NavigationHeader.propTypes = {
  iOSStatusBarColor: PropTypes.string,
};

NavigationHeader.defaultProps = {
  iOSStatusBarColor: intenseBlue,
};

const buttonWrap = {
  width: 60,
  height: 40,
  justifyContent: 'center',
  alignItems: 'center',
};

const styles = StyleSheet.create({
  container: {
    paddingLeft: 25,
    flexDirection: 'row',
    height: headerHeight,
    alignItems: 'center',
    backgroundColor: intenseBlue,
    justifyContent: 'space-between',
  },
  titleText: {
    fontSize: 17,
    color: 'white',
  },
  rightView: {
    ...buttonWrap,
  },
});
