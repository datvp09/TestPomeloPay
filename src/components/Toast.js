import {ToastAndroid, Alert} from 'react-native';
import {isiOS} from '../Constants';

export default function Toast(message) {
  if (isiOS) {
    Alert.alert(message);
  } else {
    ToastAndroid.show(message, ToastAndroid.SHORT);
  }
}
