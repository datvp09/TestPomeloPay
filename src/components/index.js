import NavigationHeader from './NavigationHeader';
import Toast from './Toast';

export {NavigationHeader, Toast};
