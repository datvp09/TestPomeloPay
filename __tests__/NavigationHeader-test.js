import 'react-native';
import React from 'react';
import {NavigationHeader} from '../src/components';
import renderer from 'react-test-renderer';

describe('TripInfoItem', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer
        .create(
          <NavigationHeader
            style={{height: 60}}
            title={'All Transactions'}
            rightView={() => {}}
          />,
        )
        .toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
