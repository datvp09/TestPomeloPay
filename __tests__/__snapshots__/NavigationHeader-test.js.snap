// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripInfoItem Rendering should match to snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1365AF",
        "height": 20,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1365AF",
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "space-between",
          "paddingLeft": 25,
        },
        Object {
          "height": 60,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 17,
          },
          undefined,
        ]
      }
    >
      All Transactions
    </Text>
  </View>
</View>
`;
